packages/sequence-diagram/src/validateDiagram.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  ValidationResult,
+  Actor,
+  Action,
+  isLoop,
+  NodeType,
+  isFunction,
+  isServerRPC,
+  isClientRPC,
+  isQuery,
+} from './types';
+
+export default async function validateDiagram(diagramData: any): Promise<ValidationResult> {
+  // Check if it's likely an AppMap object
+  if (
+    diagramData &&
+    diagramData.hasOwnProperty('metadata') &&
+    diagramData.hasOwnProperty('classMap') &&
+    diagramData.hasOwnProperty('events')
```

Comment:
If you run `yarn lint` from `packages/sequence-diagram`, you'll see that the linter throws an error. Instead of using `hasOwnProperty`, please just directly attempt to access the property (i.e. `diagramData.metadata`).

Reply:
Directly trying to access leads to typescript errors. Will instead use `Object.prototype.hasOwnProperty.call(foo, "bar");`